packages/react-components/src/components/OptionsButton.tsx

```diff
@@ -19,15 +19,15 @@ export interface OptionsButtonProps extends IButtonProps {
   /**
    * Utility props for stateful props.
    */
-  microphones?: [{ id: string; name: string }];
-  speakers?: [{ id: string; name: string }];
-  cameras?: [{ id: string; name: string }];
+  microphones?: { id: string; name: string }[];
+  speakers?: { id: string; name: string }[];
+  cameras?: { id: string; name: string }[];
   selectedMicrophone?: { id: string; name: string };
   selectedSpeaker?: { id: string; name: string };
   selectedCamera?: { id: string; name: string };
-  onSelectCamera?: (device: any) => Promise<void>;
-  onSelectMicrophone?: (device: any) => Promise<void>;
-  onSelectSpeaker?: (device: any) => Promise<void>;
+  onSelectCamera?: (device: any) => void;
+  onSelectMicrophone?: (device: any) => void;
+  onSelectSpeaker?: (device: any) => void;
```

Comment:
What is the reason for this change? Is it to fix some compiler issues? I believe the handlers for onSelectCamera, onSelectMicrophone, and onSelectSpeaker return Promise<void> , I'm wondering if we can keep it the same.

Reply:
Yeah no problem. No compiler issues with keeping it the same. Interestingly, usePropsFor will still provide the handlers with the type (device: any) => void. I just thought a return type of void is less specific and allow more flexibility to use OptionsButton. But maybe we intend to use OptionsButton with our selectors.